src/pages/index.js

```diff
@@ -39,38 +42,183 @@ const profFormPopup = new PopupWithForm(
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    handleCardEnlargement,
+    handleTrashClick,
+    sendLikeReq
+  );
   return cardElement.render();
 }
+api
+  .getCards()
+  .then((data) => {
+    data.reverse();
+    const initialSection = new Section(
+      { items: data, renderer: createCard },
+      ".gallery"
+    );
 
-section.renderItems();
+    initialSection.renderItems();
+  })
+  .catch((err) => console.error(err));
 
 // ENLARGE CARD IMAGE
 function handleCardEnlargement(evt) {
   enlrgImgPopup.open(evt);
 }
 
 // SUBMIT NEW CARD MODAL
-export function handleNewCardFormSubmit(newCardData) {
-  section.addItem(createCard(newCardData));
+function handleNewCardFormSubmit({ name, link }) {
+  const newCardApi = new Api({
+    method: "POST",
+    headers: apiHeaders,
+    body: JSON.stringify({
+      name,
+      link,
+    }),
+  });
 
-  newCardFormPopup.close();
-  newCardModalForm.reset();
+  newCardFormPopup.popup.querySelector(".modal__submit-button").textContent =
+    "Creating...";
 
-  newCardFormValidator.resetValidation();
+  newCardApi
+    .getCards()
+    .then((data) => {
+      const newCardSection = new Section(
+        { items: [data], renderer: createCard },
+        ".gallery"
+      );
+      newCardSection.renderItems();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      newCardFormPopup.popup.querySelector(
+        ".modal__submit-button"
+      ).textContent = "Create";
+      newCardFormPopup.close();
+      newCardModalForm.reset();
+      newCardFormValidator.resetValidation();
+    });
 }
 
 // SUBMIT PROFILE MODAL
-export function handleProfileSubmit(newProfData) {
-  profUserInfo.setUserInfo(newProfData);
-  profFormPopup.close();
+function handleProfileSubmit({ name, description: about }) {
+  const editProfileApi = new Api({
+    method: "PATCH",
+    headers: apiHeaders,
+    body: JSON.stringify({
+      name,
+      about,
+    }),
+  });
+  profFormPopup.popup.querySelector(".modal__submit-button").textContent =
+    "Saving...";
+  editProfileApi
+    .getUserInfo()
+    .then(() => {
+      profName.textContent = profNameInput.value;
+      profDescr.textContent = profDescrInput.value;
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      profFormPopup.popup.querySelector(".modal__submit-button").textContent =
+        "Save";
+      profFormPopup.close();
+    });
+}
+
+// DELETE CARD MODAL
+function handleTrashClick(evt) {
+  delCardFormPopup.open();
+  delCardFormPopup.popup.querySelector("input").value =
+    evt.target.closest(".gallery__card").id;
+}
+
+function handleDelCardSubmit({ cardId }) {
+  delCardApi
+    .deleteCardInfo(cardId)
+    .then(() => {
+      document.getElementById(cardId).remove();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(delCardFormPopup.close());
```

Comment:
`    .finally(delCardFormPopup.close());
`
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well.